docs/classes.md

```diff
@@ -0,0 +1,428 @@
+  
+```vb
+function __Duck_Build
+    instance = {}
+    return instance
+end function
+function Duck(name as string)
+    instance = __Duck_Build()
+    return instance
+end function
+
+function __BabyDuck_Build(name as string, age as integer)
+    instance = __Duck_Build()
+    instance.age = age
+end function
+function BabyDuck(name as string, age as integer)
+    instance = __BabyDuck_Build()
+    return instance
+end function
+```
+</details>
+
+### Calling parent method from child
+You can also call the original methods on the base class from within an overridden method on a child class. 
+
+```vb
+class Duck
+    public sub walk(meters as integer)
+        print "Walked " + meters.ToStr() + " meters"
+    end sub
+end class
+
+class BabyDuck extends Duck
+    public override sub walk(meters as integer)
+        print "Tripped"
+        super.walk(meters)
+    end sub
+end class
+
+```
+<details>
+  <summary>View the transpiled BrightScript code</summary>
+  
+```vb
+function __Duck_Build
+    instance = {}
+    instance.walk = sub(meters as integer)
+        print "Walked " + meters.ToStr() + " meters"
+    end sub
+    return instance
+end function
+function Duck(name as string)
+    instance = __Duck_Build()
+    return instance
+end function
+
+function __BabyDuck_Build(name as string, age as integer)
+    instance = __Duck_Build()
+    instance.super0_Walk = instance.walk
+    instance.walk = sub(meters as integer)
+        print "Tripped"
+        m.super0_walk(meters)
+    end sub
+end function
+function BabyDuck(name as string, age as integer)
+    instance = __BabyDuck_Build()
+    return instance
+end function
+```
+</details>
+
+## Public by default
+Class fields and methods are public by default, which aligns with the general BrightScript approach that "everything is public". 
+
+```vb
+class Person
+    'defaults to public
+    name as string
+
+    'specified private
+    private socialSecurityNumber
+
+    'defaults to public
+    function getName()
+        return m.name
+    end function
+
+    'specified private
+    private sub setSocialSecurityNumber(value as string)
+        m.socialSecurityNumber = value
+    end sub
+end class
+```
+<details>
+  <summary>View the transpiled BrightScript code</summary>
+  
+```vb
+function Person()
+    instance = {}
+    'defaults to public
+    instance.name = invalid,
+    
+    'specified private
+    instance.socialSecurityNumber = invalid
+    
+    'defaults to public
+    instance.getName = function()
+        return m.name
+    end function
+
+    'specified private
+    instance.setSocialSecurityNumber = sub(value as string)
+        m.socialSecurityNumber = value
+    end sub
+    return instance
+end function
+```
+</details>
+
+You will get compile-time errors whenever you access private members of a class from outside the class. However, be aware that this is only a compile-time restriction. At runtime, all members are public.
```

Comment:
what is the position on extended properties? I see no protected keyword (I think I have submitted an issue for that, if not I shall); but for now, do we need to be explicit about how you intend  private visibility to work with inheritance?

Reply:
The position is, they don't exist. A child class and super class will share all of the same properties. The compiler will add an error if a child and parent declare a property with the same name.